src/Core/Grid/DataProvider/GridDataProviderInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\DataProvider;
+
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Interface GridDataProviderInterface defines contract for grid data providers
+ */
+interface GridDataProviderInterface
+{
+    /**
+     * Get filtered & paginated rows from any data source (database, API or any other)
+     *
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return array
```

Comment:
it's not array but RowsCollection imo

Reply:
probably right, RowsCollection should be wrapper around raw rows returned from db?

Reply:
yes :) let's say we don't want to manipulate untyped arrays in this Grid component